iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeRequestOtpView.swift

```diff
@@ -0,0 +1,202 @@
+//  are permitted provided that the following conditions are met:
+//  * Redistributions of source code must retain the above copyright notice, this list of conditions
+//  and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright notice, this list of
+//  conditions and the following disclaimer in the documentation and/or other materials provided
+//  with the distribution.
+//  * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+//  endorse or promote products derived from this software without specific prior written
+//  permission of salesforce.com, inc.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+//  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+//  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import SalesforceSDKCore
+import SwiftUI
+
+///
+/// A view enabling the user to start the Salesforce Identity API's "Headless Passwordless Login Flow for
+/// Public Clients."  This view corresponds to the `init/passwordless/login` endpoint by collecting
+/// a Salesforce username and the user's chosen OTP verification method of email or SMS.
+///
+/// The implementation of the API endpoint's client is provided by Salesforce Mobile SDK in the native login
+/// manager.
+///
+/// On receiving a successful response from the endpoint, a separate view is used to finish the flow.
+///
+struct NativeRequestOtpView: View {
+    
+    @Environment(\.colorScheme) var colorScheme
+    
+    /// The Google reCAPTCHA client.
+    @EnvironmentObject var reCaptchaClientObservable: ReCaptchaClientObservable
+    
+    /// The navigation path.
+    @EnvironmentObject var navigationPathObservable: NavigationPathObservable
+    
+    /// An error message displayed to the user when needed.
+    @State private var errorMessage = ""
+    
+    /// An authenticating state to determine the progress indicator.
+    @State private var isAuthenticating = false
+    
+    /// The user's chosen OTP verification method of email or SMS.
+    @State private var otpVerificationMethod = OtpVerificationMethod.sms
+    
+    /// The user's entered username.
+    @State private var username = ""
+    
+    var body: some View {
+        VStack {
+            Spacer()
+            
+            HStack { Spacer() }
```

Comment:
Why does this Spacer need an HStack?

Reply:
I picked that up from `NativeLoginView`.  It seems like that was there to make the view fill the horizontal width of its parent.  It sure looked like it was to spacers: One for the vertical and one for the horizontal (Like the Twilight Zone intro).

Reply:
you might be able to cut it and the two spacers before/after the ZStack if you set something like this on the VStack
`.frame(maxWidth: .infinity, maxHeight: .infinity)`

Reply:
I tried that, then realized it's that the content will "size-to-fit" centered as the rest of the layout I borrowed from native login view stands.

<img width="545" alt="Screenshot 2024-03-19 at 17 06 46" src="https://github.com/forcedotcom/SalesforceMobileSDK-Templates/assets/94090596/0e897244-9234-4a55-9ba9-d8f5235b2f2e">

The spacers are instructing the view to fill its parent edge-to-edge so it doesn't look like the above.  The native login view on dev currently wants to look more like the below.

<img width="526" alt="Screenshot 2024-03-19 at 17 09 18" src="https://github.com/forcedotcom/SalesforceMobileSDK-Templates/assets/94090596/ab6e1ac8-2e11-4d0d-9401-da33adcc4f33">

Reply:
That said, I see in the docs the modifier you sent _should_ have worked.  I'm looking again 🧐

Reply:
@bbirman - This all got set back to the state on `dev` in this morning's commits.